app.py

```diff
@@ -37,7 +37,7 @@ def submit():
         else:
             # User Submits Job #
             # (1) Create unique ID for each submission
-            timestamp = datetime.datetime.now().strftime('%Y%m%d%H%M%S%f')
+            timestamp = datetime.now().strftime('%Y%m%d%H%M%S%f')
```

Comment:
Are you sure about this?

```python
>>> import datetime
>>> datetime.now().strftime('%Y%m%d%H%M%S%f')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: module 'datetime' has no attribute 'now'
>>> datetime.datetime.now().strftime('%Y%m%d%H%M%S%f')
'20240530144003531169'
```

Reply:
Hi Professor. I just restart production site to tested it once and didn't find this problem. 
Here is worker.out.log: 
########################################
[05/30/24 14:48:12] START 20240530144811740349
########################################
mkdir -p ././uploads/20240530144811740349
Downloading ftp://ftp.ensembl.org/pub/release-88/fasta/mus_musculus/dna/Mus_musculus.GRCm38.dna.toplevel.fa.gz
pigz -d ././uploads/20240530144811740349/Mus_musculus.GRCm38.dna.toplevel.fa.gz
Downloading ftp://ftp.ensembl.org/pub/release-88/gff3/mus_musculus/Mus_musculus.GRCm38.88.gff3.gz
pigz -d ././uploads/20240530144811740349/Mus_musculus.GRCm38.88.gff3.gz
mkdir -p ./results/20240530144811740349
/home/reform/venv/bin/python reform.py --chrom 1 --position 0 --in_fasta ./uploads/20240530144811740349/in.fa --in_gff ./uploads/20240530144811740349/in.gtf --ref_fasta ././uploads/20240530144811740349/Mus_musculus.GRCm38.dna.toplevel.fa --ref_gff ././uploads/20240530144811740349/Mus_musculus.GRCm38.88.gff3 --output_dir ./results/20240530144811740349/
Checking position validity
Position valid
Proceeding to insert sequence 'test 1' from ./uploads/20240530144811740349/in.fa at position 0 on chromsome 1
New fasta file created:  ./results/20240530144811740349/Mus_musculus.GRCm38.dna.toplevel_reformed.fa
Preparing to create new annotation file
New .GFF3 file created: ./results/20240530144811740349/Mus_musculus.GRCm38.88_reformed.gff3 
rm -Rf ./uploads/20240530144811740349
mkdir -p ./downloads/20240530144811740349
tar cf - ./results/20240530144811740349/ | pigz  > ./downloads/20240530144811740349/reformed.tar.gz
rm -Rf ./results/20240530144811740349
########################################
[05/30/24 15:05:48] END 20240530144811740349
########################################
Emailing
Emailed


And here is worker.err.log:
2024-05-30 14:49:16 URL: ftp://ftp.ensembl.org/pub/release-88/fasta/mus_musculus/dna/Mus_musculus.GRCm38.dna.toplevel.fa.gz [826868288] -> "././uploads/20240530144811740349/Mus_musculus.GRCm38.dna.toplevel.fa.gz" [1]
2024-05-30 14:49:59 URL: ftp://ftp.ensembl.org/pub/release-88/gff3/mus_musculus/Mus_musculus.GRCm38.88.gff3.gz [59857040] -> "././uploads/20240530144811740349/Mus_musculus.GRCm38.88.gff3.gz" [1]

Reply:
Don't understand why it works when ran in a script but not manually. I now attempted as reform user and with the venv.